tests/test_basic_doc.py

```diff
@@ -1,70 +1,22 @@
 import json
 import os.path
-import re
 import sys
 import tempfile
-import uuid
 from pathlib import Path
-from random import randrange
 
 import pytest
-import responses
 from sphinx import version_info
 from sphinx.application import Sphinx
 from sphinx.testing.util import SphinxTestApp
 from syrupy.filters import props
 
 from sphinx_needs.api.need import NeedsNoIdException
-from tests.data.service_github import (
-    GITHUB_ISSUE_SEARCH_ANSWER,
-    GITHUB_SEARCH_COMMIT_ANSWER,
-    GITHUB_SPECIFIC_COMMIT_ANSWER,
-    GITHUB_SPECIFIC_ISSUE_ANSWER,
-)
-
 
-def random_data_callback(request):
-    """
-    Response data callback, which injects random ids, so that the generated needs get always a unique id and no
-    exceptions get thrown.
-    """
-    if re.match(r"/search/issues", request.path_url):
-        data = GITHUB_ISSUE_SEARCH_ANSWER
-        data["items"][0]["number"] = randrange(10000)
-    elif re.match(r"/.+/issue/.+", request.path_url) or re.match(
-        r"/.+/pulls/.+", request.path_url
-    ):
-        data = GITHUB_SPECIFIC_ISSUE_ANSWER
-        data["number"] = randrange(10000)
-    elif re.match(r"/search/commits", request.path_url):
-        data = GITHUB_SEARCH_COMMIT_ANSWER
-        data["number"] = randrange(10000)
-    elif re.match(r"/.*/commits/*", request.path_url):
-        data = GITHUB_SPECIFIC_COMMIT_ANSWER
-        data["sha"] = uuid.uuid4().hex
-    else:
-        print(request.path_url)
-    return 200, [], json.dumps(data)
 
-
-#  OFFICIAL DOCUMENTATION BUILDS
-
-
-@responses.activate
 @pytest.mark.parametrize(
     "test_app", [{"buildername": "html", "srcdir": "doc_test/doc_basic"}], indirect=True
 )
 def test_build_html(test_app):
-    responses.add_callback(
```

Comment:
Does removing this mean, our tests are firing now real requests to the GitHub API.
Or have I missed the locations, where this is still mocked?

Reply:
`doc_test/doc_basic` has no `needservice` directives in, i.e. there is no requests to make 😄

Reply:
argh sure, thought these removals had some functional reason.
But now it makes all sense...